packages/flutter/lib/src/material/text_field.dart

```diff
@@ -1149,6 +1151,17 @@ class _TextFieldState extends State<TextField> with RestorationMixin implements
     }
   }
 
+  /// Sets the cursor color to error color from decoration or from theme colorScheme if [_hasError] is true else
+  /// it will return the color from [widget.cursorColor] param if not null. if [widget.cursorColor] is null then it will return default style else return
+  /// color from theme colorScheme.
+  Color _getCursorColor(DefaultSelectionStyle selectionStyle, ThemeData theme){
+    if(_hasError){
+      return _errorColor;
```

Comment:
Nit: You could just inline this here rather than making it a getter, but either way works.

Reply:
I have made it inline